test/api/v3/integration/groups/GET-groups.test.js

```diff
@@ -236,15 +236,27 @@ describe('GET /groups', () => {
       await expect(user.get('/groups?type=publicGuilds&paginate=true&page=1'))
         .to.eventually.have.a.lengthOf(GUILD_PER_PAGE);
       const page2 = await expect(user.get('/groups?type=publicGuilds&paginate=true&page=2'))
-        .to.eventually.have.a.lengthOf(1 + 4); // 1 created now, 4 by other tests
-      expect(page2[4].name).to.equal('guild with less members');
+        // 1 created now, 4 by other tests, -1 for no more tavern.
+        .to.eventually.have.a.lengthOf(1 + 4 - 1);
+      expect(page2[3].name).to.equal('guild with less members');
     }).timeout(10000);
   });
 
+  it('makes sure that the tavern doesn\'t show up when guilds is passed as a query', async () => {
```

Comment:
for this and the next test could we also add a check that `guilds.length` is `0` so that we're sure there's only `guilds[0]` to check?

Reply:
Because of previous tests, `guilds.length` at that moment is 4 and 2 for type `guilds` and `publicGuilds` respectively.

This seems to be because in the `before` some guilds already got created for other tests, like this one:
https://github.com/HabitRPG/habitica/blob/13a5b276e94e3e571e9a5d1fc641d5d0c82ac138/test/api/v3/integration/groups/GET-groups.test.js#L28-L40

So the two options are then to loop over every single guild and check if they don't contain `TAVERN_ID`, or to reset the database again using `await resetHabiticaDB();` to make sure that there are actually zero guilds.

The first has the downside that it could make the tests take unnecessary long in the future, if more new guilds end up being created. The second might cause the tests afterwards to fail, so the tests would have to be moved to the end instead.

Which option would be the wisest to go for in this regard? Personally I think `await resetHabiticaDB();` is the cleaner option, since it makes sure that the Tavern documents becomes part of the guilds before the test.

Reply:
I'd go with the first option and loop over the resulting array to make sure the tavern isn't there, there are few guilds and even if some are added it would take thousands of results to slow down the tests.

```
expect(guilds.find(g => g.id === TAVERN_ID)).to.be.undefined
```

^ something like this should work!